pxf/pxf-json/src/main/java/org/apache/hawq/pxf/plugins/json/JsonStreamReader.java

```diff
@@ -0,0 +1,121 @@
+package org.apache.hawq.pxf.plugins.json;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * Reads a JSON stream and sequentially extracts all JSON objects identified by the <b>identifier</> parameter. Returns
+ * null when there are no more objects to read.
+ *
+ */
+public class JsonStreamReader extends BufferedReader {
+
+	private static char START_BRACE = '{';
+	private static char END_BRACE = '}';
+	private static int EOF = -1;
+
+	private StringBuilder bldr = new StringBuilder();
+	private String identifier = null;
+	private long bytesRead = 0;
+
+	/**
+	 * @param identifier
+	 *            JSON object name to extract.
+	 * @param inputStream
+	 *            JSON document input stream.
+	 */
+	public JsonStreamReader(String identifier, InputStream inputStream) {
+		super(new InputStreamReader(inputStream));
+		this.identifier = identifier;
+	}
+
+	/**
+	 * @return Returns next JSON object identified by the {@link JsonStreamReader#identifier} parameter or Null if no
+	 *         more object are available.
+	 * @throws IOException
+	 */
+	public String getJsonRecord() throws IOException {
+		bldr.delete(0, bldr.length());
+
+		boolean foundRecord = false;
+
+		int c = 0, numBraces = 1;
+		while ((c = super.read()) != EOF) {
+			++bytesRead;
+			if (!foundRecord) {
+				bldr.append((char) c);
+
+				if (bldr.toString().contains(identifier)) {
+					if (forwardToStartBrace()) {
+						foundRecord = true;
+
+						bldr.delete(0, bldr.length());
+						bldr.append(START_BRACE);
+					}
+				}
+			} else {
+				bldr.append((char) c);
+
+				if (c == START_BRACE) {
+					++numBraces;
+				} else if (c == END_BRACE) {
+					--numBraces;
+				}
+
+				if (numBraces == 0) {
+					break;
+				}
+			}
+		}
+
+		if (foundRecord && numBraces == 0) {
+			return bldr.toString();
+		} else {
+			return null;
```

Comment:
yes it makes sense and i will add it. But mind that the current stream reader is not aware of the semantic of the json document and can miss chunks from the documents if later is  malformed. We need to implement a proper stream reading to minimize such issues.